tests/test_modeling_common.py

```diff
@@ -410,7 +410,7 @@ def test_forward_signature(self):
                 )
                 self.assertListEqual(arg_names[: len(expected_arg_names)], expected_arg_names)
             else:
-                expected_arg_names = ["input_ids"]
+                expected_arg_names = [model.main_input_name]
```

Comment:
I like this very much but I fear there may be some edge cases. 

What if we had a list in the class, e.g. `expected_arg_names` and we compose them 

```python
expected_arg_names = [model.main_input_name] + self.expected_arg_names 

```

`self.expected_arg_names` is initialized with `None`or with whatever the user passes